src/tableDrawer.ts

```diff
@@ -34,19 +35,73 @@ export function drawTable(jsPDFDoc: jsPDFDocument, table: Table): void {
   const startPos = assign({}, cursor)
 
   table.startPageNumber = doc.pageNumber()
+  console.log('inside splitColumns - 0 ****** ', settings)
 
-  doc.applyStyles(doc.userStyles)
-  if (settings.showHead === 'firstPage' || settings.showHead === 'everyPage') {
-    table.head.forEach((row) => printRow(doc, table, row, cursor))
-  }
-  doc.applyStyles(doc.userStyles)
-  table.body.forEach((row, index) => {
-    const isLastRow = index === table.body.length - 1
-    printFullRow(doc, table, row, isLastRow, startPos, cursor)
-  })
-  doc.applyStyles(doc.userStyles)
-  if (settings.showFoot === 'lastPage' || settings.showFoot === 'everyPage') {
-    table.foot.forEach((row) => printRow(doc, table, row, cursor))
+  if (settings.splitColumns === true) {
+    // managed flow for split columns
+    // calculate width of columns and render only those which can fit into page
+    const allColumnsCanFitResult: ColumnFitInPageResult[] = tablePrinter.calculateAllColumnsCanFitInPage(
+      doc,
+      table
+    )
+
+    allColumnsCanFitResult.map(
+      (colsAndIndexes: ColumnFitInPageResult, index: number) => {
+        doc.applyStyles(doc.userStyles)
+        // add page to print next columns in new page
+        if (index > 0) {
+          addPage(doc, table, startPos, cursor, {
+            columns: colsAndIndexes.columns,
+          })
+        } else if (
+          settings.showHead === 'firstPage' ||
+          settings.showHead === 'everyPage'
+        ) {
+          // print head for selected columns
+          table.head.forEach((row) =>
+            printRow(doc, table, row, cursor, {
+              columns: colsAndIndexes.columns,
+            })
+          )
+        }
+        // print body for selected columns
+        doc.applyStyles(doc.userStyles)
+        table.body.forEach((row, index) => {
+          const isLastRow = index === table.body.length - 1
+          printFullRow(doc, table, row, isLastRow, startPos, cursor, {
```

Comment:
Instead of passing config it might be cleaner to pass the columns we want to draw to the printFullRow and printRow method right away? Even if it is all columns (table.columns). We should avoid any parameters as well.